### ISSUE TITLE
LHN-Reports/chats missing in LHN when logging in to account in "#focus" priority mode

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#43612]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.82-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y 
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

## Action Performed:
1. Log in to New Expensify with account applausetester+ak6122nd@applause.expensifail.com
2. Note the amount of reports in LHN
3. Navigate to settings > Preferences
4. Change priority mode to "#focus"
5. Log out and log in with the same account
5. Change the priority mode back to Most Recent

## Expected Result:
The same amount of chats should be displayed as in Step 2

## Actual Result:
Many chats (depending on the amount in account) are missing in LHN
The chats do not reappear after refreshing the page/relaunching the app, user is forced to change priority mode to most recent and relog to see all reports in LHN
Not reproducible for pinned/GBR chats.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/5534c0c6-13a8-4d13-a081-9c17780c3ec3

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~013c1484a9570e0a56</li>
        <li>Upwork Job ID: 1801290617630180667</li>
        <li>Last Price Increase: 2024-07-15</li>
    </ul>
</details>

**Comment by lanitochka17** [comment]
@twisterdotcom  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by twisterdotcom** [comment]
Waiting on proposals

**Comment by twisterdotcom** [comment]
Waiting on proposals

**Comment by getusha** [comment]
Waiting for proposals

**Comment by twisterdotcom** [comment]
Doubling

**Comment by twisterdotcom** [comment]
Still waiting on proposals.

**Comment by twisterdotcom** [comment]
Okay, I'm not clear if this is still an issue. I tested it with pecan@expensivepie.com and couldn't recreate. Gonna ask for a retest.

**Comment by mvtglobally** [comment]
Issue is reproducible during KI retests. 

https://github.com/Expensify/App/assets/43995119/06a72e13-eaf7-4f3f-9675-14846e4fbc97

**Comment by twisterdotcom** [comment]
Okay, so the recreation steps should be amended a bit:

1. Log in to any account
2. Make the preference "Most recent"
3. Pin a few of the chats, but leave some unpinned and without drafts
4. Switch to Focus mode
5. Sign in to the same account in a different browser window
6. Switch back to "Most recent"
7. See we did not load the unpinned chats into the LHN.

**Comment by Kureev** [comment]
Hi, I am Alexey from Margelo and I can work on this issue.

**Comment by Kureev** [comment]
This one is likely to be a BE issue: when we switch the priority mode in preferences, the request is sent to BE to update chat priority mode. The request seems correct and work properly when we switch modes from default to focus and back, however, if we switch the mode to focus, log out, log in again and switch it back to default (see ticket description), it seems that the next OpenReport query returns old data (3 records instead of 5 in my case), even though the flag has been changed

**Comment by twisterdotcom** [comment]
@getusha do you agree?

**Comment by getusha** [comment]
Yes, it looks like a backend issue. I see that `nvp_priorityMode` is set to the correct value, but as @Kureev mentioned, `OpenReport` is returning the filtered data despite the change.

**Comment by twisterdotcom** [comment]
Okay, let's make this internal.

**Comment by muttmuure** [comment]
I've just realized. This isn't a quality issue

**Comment by muttmuure** [comment]
It's a regular bug, because it's reliably reproduced.

**Comment by muttmuure** [comment]
I'm going to remove it from the project (sorry!)

**Comment by twisterdotcom** [comment]
Okay, this is still `Internal` and super annoying because it means your LHN is out of sync. Seems we know the problem, we just need somebody internal to take it on.

**Comment by twisterdotcom** [comment]
No internal assignee yet. Going to downgrade to `Monthly` given no uptake.

**Comment by twisterdotcom** [comment]
Not a priority it seems then
